spec/models/create_cap_proposal_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe CreateCapProposal do
+  let(:auction) { FactoryGirl.create(:auction, :with_bidders, :evaluation_needed) }
+  let(:auction_presenter) { Presenter::Auction.new(auction) }
+  let(:create_cap_proposal) { CreateCapProposal.new(auction_presenter) }
+  let(:c2_proposal_hash) { create_cap_proposal.c2_proposal_hash }
+
+  describe '#c2_proposal_hash' do
```

Comment:
make #c2_proposal_hash private, and test #perform instead. Do something like

```
expect(c2_client).to receive(:post).with(proposals', c2_proposal_hash_stub).and_return(mocked_response)
```

....